### ISSUE TITLE
Adding/Removing from watchlist not working

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#10940]
## Bug description 

User reports incoming that adding/removing a cache to/from watchlist is no longer working

### Steps to reproduce the problem

- Open cache details
- Select "+" to add the cache to watchlist or "-" to remove it from watchlist

### Actual result after these steps 

Error toast "Updating watchlist not successful" (or similar)

### Expected result after these steps 

Changing watchlist state OK

## c:geo version 

2021.05.120 and current `master`

## Reproducible 

Yes

## Additional context 

Related to #10939

**Comment by eddiemuc** [comment]
I am very confident that this has the same root cause as #10939

**Comment by balu2003** [comment]
I think that's another bug related to the watchlist. If you receive an e-mail about a listing because you are on the watchlist, then at the end of the e-mail there is a link "Remove this from my watchlist". The link in the e-mail has this URL using the example: "http://www.geocaching.com/my/watchlist.aspx?id=3719606&DS=1&ACTION=REM"
But If you click on theís  link in the e-mail, the cache will NOT be removed from the watchlist., (This is a bug from gc website IMHO)
-- **I Think that cgeo use the above URL , which ist not working right now**

However, if you click on "Watch" or "Stop Watching" on the website, the URL "https://www.geocaching.com/seek/cache_details.aspx/HandleWatchlistAction" is called and a JSON payload {true | false} is passed. The Cache will be added | removed from the watchlist.
This is new and has not been the case so far and I suspect by changing the website scripts it was forgotten that there is also the link "Remove this from ..". However, it no longer works.
(sry bad translation) 
 
```
Part of the request header:
POST /seek/cache_details.aspx/HandleWatchlistAction HTTP/1.1
Host: www.geocaching.com
...
Content-Type: application/json; charset=UTF-8
Origin: https://www.geocaching.com
...
Referer: https://www.geocaching.com/geocache/GC4EMB7_4000x-agent230-ein-lpc-und-doch-keiner
 ...
with JSON Payload Add = {true | false}
  ...
see the html src of a cache Listing:
function addToWatchlist(event) 
function removeFromWatchlist(event) {
function HandleWatchlistAction($self, isAdd) {
```

**Comment by eddiemuc** [comment]
Fix is provided ion PR #10997
